lib/crypto/key.go

```diff
@@ -18,17 +18,21 @@ func ReadPub64(suite abstract.Suite, r io.Reader) (abstract.Point, error) {
 // WritePub64 writes a public point to a base64 representation
 func WritePub64(suite abstract.Suite, w io.Writer, point abstract.Point) error {
 	enc := base64.NewEncoder(base64.StdEncoding, w)
-	err := suite.Write(enc, point)
-	enc.Close()
-	return err
+	return write64(suite, enc, point)
+}
+
+func write64(suite abstract.Suite, wc io.WriteCloser, data ...interface{}) error {
+	errW := suite.Write(wc, data)
+	if err := wc.Close(); err != nil && errW == nil {
+		return err
+	}
+	return errW
```

Comment:
I remember why I did this this way. If you do
```
if err := suite.Write(wc, data); err != nil {
               return err
}
``` 
and return in the first `if`: I am not sure if the writer is opened and stays open (`Close` won't be called).